include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -1165,25 +1156,23 @@ namespace internal
       (void)dummy2;
     }
 
-    template <int direction, bool contract_over_rows, bool add>
+    template <int direction, bool contract_over_rows, bool add, int stride = 1>
```

Comment:
This is the relevant change to support placing the gradients differently: We need to be able to set a stride, typically `dim`, on one of the arrays.

Reply:
There seems to be no comment about what stride is in this context? Stride within input/output vector?

Reply:
I added a comment, commit coming soon.